cdap-unit-test/src/test/java/io/cdap/cdap/test/app/AppWithCustomTx.java

```diff
@@ -427,12 +430,20 @@ public ByteBuffer nextChunk(Transactional transactional) throws Exception {
             public void onFinish() throws Exception {
               recordTransaction(getContext(), PRODUCER_TX, DESTROY);
               attemptNestedTransaction(getContext(), PRODUCER_TX, DESTROY_NEST);
+              File doneFile = new File(getContext().getRuntimeArguments().get(DONE_DIR), "tx" + body);
```

Comment:
can you add some comments about onFinish and onError being called after the connection is closed so that it is clear why this is being done.